src/openms_gui/source/VISUAL/TOPPViewMenu.cpp

```diff
@@ -90,7 +90,17 @@ namespace OpenMS
     m_file->addMenu(recent_files->getMenu()); // updates automatically via RecentFilesMenu class, since this is just a pointer
 
     m_file->addSeparator();
-    m_file->addAction("&Preferences", parent, &TOPPViewBase::preferencesDialog);
+    
+    //Specifically set the role of the Preferences item. Additionally we have to avoid adding other action items that are
+    // called preferences/config/options and have the default TextHeuristicRole because otherwise they will overwrite the macOS specific
+    // menu entry under Application -> Preferences...
+    //m_file->addAction("&Preferences", parent, &TOPPViewBase::preferencesDialog);
```

Comment:
AFAIK `addAction` returns a `QAction*` on which you can operate
thus you can shorten to

```
auto pref = m_file->addAction("&Preferences", parent, &TOPPViewBase::preferencesDialog);
pref->setMenuRole(QAction::PreferencesRole);
pref->setEnabled(true);
```
?